contributors/design-proposals/security-model.md

```diff
@@ -0,0 +1,190 @@
+# Requirements
+
+This document considers "customer facing" entities. For our purposes these are
+the Kubernetes API, and running workloads.
+
+1. Kubernetes needs to be safe in a multi-tenant environment
+2. Securing Kubernetes should be easy
+
+This document uses the "[CIA triad](https://en.wikipedia.org/wiki/Information_security#Key_concepts)" for analyzing these goals.
+
+# Kubernetes API access controls
+
+Kubernetes needs access controls capable of protecting the CIA of cluster
+resources in a multi-tenant environment.
+
+## Concerns with method based access controls
+
+Many Kubernetes deployments provide method based access control (e.g.,
+list pods). This is akin to securing \*nix with permissions on the "open"
+syscall. To allow more than privileged user, ACLs should be placed on nouns, not
+verbs.
+
+## Concerns with implicit relationships
+
+Kubernetes' implicit object relationships create subtle dangers.
+
+1. The "create pod" privilege allows a user to mount a secret, even if the user
+   cannot "get secrets". This is a failure of confidentiality.
+2. The "create pod" privilege can be used to place a pod behind a service, even
+   if the user cannot "update service". This is a failure of integrity.
+3. The "create pod" privilege can be used to place a pod behind a deployment,
+   which can be used to starve legitimate replicas. This is a failure of
+   integrity, and availability.
+
+This isn't a complete list of such dangers, but it's sufficient to show that
+access controls on an object are insufficient to guarantee CIA. Access controls
+are also needed on the relationships between objects.
+
+## Label ACLs (implicit ownership)
+
+Label ACLs are sometimes suggested as a method to provide access control on
+object-object relationships. Special permissions would be required to create or
+query an ACL'ed label. This approach is both complex and insufficient.
+
+1. The accidental omission of a label means that the system fails open.
```

Comment:
That's true, but I dread a k8s where I need permissions for _every_ label. I don't see how access management is simplified by doing this.